Sources/ClientRuntime/Networking/Http/Middlewares/FlexibleChecksumsRequestMiddleware.swift

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
+
+public struct FlexibleChecksumsRequestMiddleware<OperationStackInput, OperationStackOutput>: Middleware {
+
+    public let id: String = "FlexibleChecksumsRequestMiddleware"
+
+    let checksumAlgorithms: [String]
+
+    public init(checksumAlgorithms: [String]) {
+        self.checksumAlgorithms = checksumAlgorithms
+    }
+
+    public func handle<H>(context: Context,
+                          input: SerializeStepInput<OperationStackInput>,
+                          next: H) async throws -> OperationOutput<OperationStackOutput>
+    where H: Handler,
+    Self.MInput == H.Input,
+    Self.MOutput == H.Output,
+    Self.Context == H.Context {
+        
+        // Initialize logger
+        guard let logger = context.getLogger() else {
+            return try await next.handle(context: context, input: input)
+        }
+
+        // Get supported list of checksums in priority order
+        let validationList = HashFunction.fromList(checksumAlgorithms).getPriorityOrderValidationList()
+        
+        // Skip flexible checksums workflow if no valid checksum algorithms are provided
+        guard let checksumAlgorithm: HashFunction = validationList.first else {
+            logger.info("Found no supported checksums! Skipping flexible checksums workflow...")
```

Comment:
Q: Should an error be thrown instead if an operation is supposed to support checksum algorithm to use? Or is providing checksum algorithm always optional?

Reply:
Will change to logging an error. No need to stop the request at this point we can just "skip the workflow" as stated in the SEP